crates/aptos/src/common/transactions.rs

```diff
@@ -0,0 +1,343 @@
+            ))
+        }
+    }
+
+    /// Retrieves the private key and the associated address
+    /// TODO: Cache this information
+    pub fn get_key_and_address(&self) -> CliTypedResult<(Ed25519PrivateKey, AccountAddress)> {
+        self.private_key_options.extract_private_key_and_address(
+            self.encoding_options.encoding,
+            &self.profile_options,
+            self.sender_account,
+        )
+    }
+
+    pub fn get_address(&self) -> CliTypedResult<AccountAddress> {
+        self.private_key_options.extract_address(
+            self.encoding_options.encoding,
+            &self.profile_options,
+            self.sender_account,
+        )
+    }
+
+    pub fn get_public_key_and_address(&self) -> CliTypedResult<(Ed25519PublicKey, AccountAddress)> {
+        self.private_key_options.extract_public_key_and_address(
+            self.encoding_options.encoding,
+            &self.profile_options,
+            self.sender_account,
+        )
+    }
+
+    pub fn sender_address(&self) -> CliTypedResult<AccountAddress> {
+        Ok(self.get_key_and_address()?.1)
+    }
+
+    pub fn get_public_key(&self) -> CliTypedResult<Ed25519PublicKey> {
+        self.private_key_options
+            .extract_public_key(self.encoding_options.encoding, &self.profile_options)
+    }
+
+    pub async fn sequence_number(&self, sender_address: AccountAddress) -> CliTypedResult<u64> {
+        let client = self.rest_client()?;
+        get_sequence_number(&client, sender_address).await
+    }
+
+    pub async fn view(&self, payload: ViewFunction) -> CliTypedResult<Vec<serde_json::Value>> {
+        let client = self.rest_client()?;
+        Ok(client
+            .view_bcs_with_json_response(&payload, None)
+            .await?
+            .into_inner())
+    }
+
+    pub async fn simulate_remotely(
+        &self,
+        payload: TransactionPayload,
+    ) -> CliTypedResult<TransactionSummary> {
+        let client = self.rest_client()?;
+        let sender_address = self.get_address()?;
+
+        let gas_unit_price = if let Some(gas_unit_price) = self.gas_options.gas_unit_price {
+            gas_unit_price
+        } else {
+            client.estimate_gas_price().await?.into_inner().gas_estimate
+        };
+
+        // Get sequence number for account
+        let (account, state) = get_account_with_state(&client, sender_address).await?;
+        let sequence_number = account.sequence_number;
+
+        // Retrieve local time, and ensure it's within an expected skew of the blockchain
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .map_err(|err| CliError::UnexpectedError(err.to_string()))?
+            .as_secs();
+        let now_usecs = now * US_IN_SECS;
+
+        // Warn local user that clock is skewed behind the blockchain.
+        // There will always be a little lag from real time to blockchain time
+        if now_usecs < state.timestamp_usecs - ACCEPTED_CLOCK_SKEW_US {
```

Comment:
They could be skewed too far the other way as well right, which might lead to them setting an expiration time further than intended and all that. Not sure if wanna log a message in that case too.

Reply:
This is copied from the other one.

I think we're mostly worried about that you're too far behind, which means your transaction won't ever go to the chain.